python/src/filesystem.rs

```diff
@@ -590,31 +580,23 @@ impl ObjectOutputStream {
 
     fn write(&mut self, data: &PyBytes) -> PyResult<i64> {
         self.check_closed()?;
-        let len = data.as_bytes().len() as i64;
         let py = data.py();
-        let data = data.as_bytes();
-        let res = py.allow_threads(|| match rt().block_on(self.writer.write_all(data)) {
```

Comment:
Did you forget re-adding this? Now the GIL will be in place

Reply:
Yep, but the write function isn't async or multithreaded (it just extends a buffer in memory) so I don't believe releasing the GIL has any impact there? It releases it in `flush` if it gets called.

Reply:
> Yep, but the write function isn't async or multithreaded (it just extends a buffer in memory) so I don't believe releasing the GIL has any impact there? It releases it in `flush` if it gets called. 

I think it's more that some folks want to run stuff in parallel/threaded and just to release the GIL whenever possible, doesn't hurt to keep it that way with allow_threads

Reply:
added it back! also figured I'd add a check in `close` to make sure we're only flushing if the buffer has data inside it.